### PR TITLE
[Issue #680 fix] Added replay functionality to Abecedarium activity

### PR DESCRIPTION
**Pull request by ksraj123**
Fixes issue #680

Added basic restart functionality to the Abecedarium activity. Now upon completion of the game a congratulations message will appear with options to either restart the same game or go to game selection.

Peek 2020-03-09 23-09

Peek 2020-03-09 23-10

### CODE REVIEW

**Comment by ksraj123** [comment]
Small changes made to styling of game finished page.
Screenshot from 2020-03-10 00-13-43

Screenshot from 2020-03-10 00-13-49

File: activities/Abecedarium.activity/play.js [hunk 1]
```diff
@@ -26,7 +26,26 @@ enyo.kind({
 			{name: "itemCount", content: "-/-", classes: "itemCount", showing: false},		
 			{name: "back", kind: "Image", src: "images/back.png", showing: false, classes: "standardButton backButton", ontap: "backTaped"},
 			{name: "filter", kind: "Image", src: "images/filter.png", showing: false, classes: "standardButton filterButton", ontap: "filterTaped"},
-			{name: "check", kind: "Image", src: "images/check.png", showing: false, classes: "standardButton checkButton", ontap: "checkTaped"}
+			{name: "check", kind: "Image", src: "images/check.png", showing: false, classes: "standardButton checkButton", ontap: "checkTaped"},
+			{name: "gameFinished", kind: "Control", showing: false, classes: "gameFinished", 
+				components: [
+					{content: "Game Finished", classes: "gameFinsihed-message"},
```
Comment (llaske) [inline activities/Abecedarium.activity/play.js]: Should be localized.

File: activities/Abecedarium.activity/play.js [hunk 2]
```diff
@@ -26,7 +26,26 @@ enyo.kind({
...
+					{classes: "gameFinished-controls", components: [
+						{tag: 'span', classes: "entry replayButton", ontap: "replayTaped",
```
Comment (llaske) [inline activities/Abecedarium.activity/play.js]: You don't need to specify ` tag: 'span' `

File: activities/Abecedarium.activity/play.js [hunk 3]
```diff
@@ -26,7 +26,26 @@ enyo.kind({
...
+				components: [
+					{content: "Game Finished", classes: "gameFinsihed-message"},
+					{classes: "gameFinished-controls", components: [
+						{tag: 'span', classes: "entry replayButton", ontap: "replayTaped", 
+							components: [
+								{classes: "gameFinished-controls-text", content: "Replay"},
+								{name: "replay", kind: "Image", src: "images/redo.svg", classes: "standardButton"}
+							]
+						},
+						{tag: 'span', classes: "entry backToSelectionBtn", ontap: "backToSelectionTaped",
```
Comment (llaske) [inline activities/Abecedarium.activity/play.js]: Same comment than previously

File: activities/Abecedarium.activity/play.js [hunk 4]
```diff
@@ -26,7 +26,26 @@ enyo.kind({
...
 			{name: "filter", kind: "Image", src: "images/filter.png", showing: false, classes: "standardButton filterButton", ontap: "filterTaped"},
-			{name: "check", kind: "Image", src: "images/check.png", showing: false, classes: "standardButton checkButton", ontap: "checkTaped"}
+			{name: "check", kind: "Image", src: "images/check.png", showing: false, classes: "standardButton checkButton", ontap: "checkTaped"},
+			{name: "gameFinished", kind: "Control", showing: false, classes: "gameFinished", 
+				components: [
+					{content: "Game Finished", classes: "gameFinsihed-message"},
+					{classes: "gameFinished-controls", components: [
+						{tag: 'span', classes: "entry replayButton", ontap: "replayTaped", 
+							components: [
+								{classes: "gameFinished-controls-text", content: "Replay"},
```
Comment (llaske) [inline activities/Abecedarium.activity/play.js]: I'm not sure a text is need

File: activities/Abecedarium.activity/play.js [hunk 5]
```diff
@@ -26,7 +26,26 @@ enyo.kind({
...
+								{name: "replay", kind: "Image", src: "images/redo.svg", classes: "standardButton"}
+							]
+						},
+						{tag: 'span', classes: "entry backToSelectionBtn", ontap: "backToSelectionTaped",
+							components: [
+								{classes: "gameFinished-controls-text", content: "Select"},
```
Comment (llaske) [inline activities/Abecedarium.activity/play.js]: Same comment than previously

File: activities/Abecedarium.activity/play.js (hunk 4 above)
Comment (ksraj123) [reply activities/Abecedarium.activity/play.js]: Just to clarify. Are you talking about the `gameFinished-controls-text` class or should I just display the icons and no Replay or Select text? Also is the chosen icon for back to selection alright? Or should I change it? Could you suggest something?

**Comment by ksraj123** [comment]
@llaske suggested changes made. Please review.
Peek 2020-03-12 00-52

**Comment by llaske** [comment]
That's better.

But your fix has changed the location of the back button.
Before:

image

After:

image

Same from the home button:

image

File: activities/Abecedarium.activity/play.js [hunk 6]
```diff
@@ -52,6 +64,9 @@ enyo.kind({
 			this.doGame(this.themeButton);
 		else
 			this.displayButtons();
+			
+		if (this.gamecount === entriesByGame)
+			this.showGameFinished();
```
Comment (llaske) [inline activities/Abecedarium.activity/play.js]: Use bracket around instructions in if/else

File: activities/Abecedarium.activity/play.js [hunk 7]
```diff
@@ -288,6 +303,34 @@ enyo.kind({
 			this.selected = entry;
 		}
 	},
+
+	showGameFinished: function(){
+		this.cleanBox();
+		Abcd.changeVisibility(this, {home: false, back: false, filter: false, check: false, itemCount: false});
+		Abcd.hideLang();
+		this.$.caseButton.hide();
+		var l10n ={"en":"Game Finished","fr":"Game Finished","es":"Game Finished","pt":"Game Finished","de":"Game Finished"};
```
Comment (llaske) [inline activities/Abecedarium.activity/play.js]: Ooch, that's not at all the right way to do that. See tutorial on localization.

File: activities/Abecedarium.activity/js/activity.js [hunk 8]
```diff
@@ -20,6 +20,7 @@ define(["sugar-web/activity/activity","sugar-web/datastore","tutorial","sugar-we
 			var defaultLanguage = (typeof chrome != 'undefined' && chrome.app && chrome.app.runtime) ? chrome.i18n.getUILanguage() : navigator.language;
 			var language = environment.user ? environment.user.language : defaultLanguage;
 			webL10n.language.code = language;
+			Abcd.languageCode = language;
```
Comment (llaske) [inline activities/Abecedarium.activity/js/activity.js]: You should not need this

File: activities/Abecedarium.activity/play.js (hunk 6 above)
Comment (ksraj123) [reply activities/Abecedarium.activity/play.js]: @llaske pre-existing code in lines above did not use brackets so I intentionally left them for uniformity in style. Will change it.

**Comment by ksraj123** [comment]
@llaske suggested changes made. Please review.

File: activities/Abecedarium.activity/play.js (hunk 7 above)
Comment (ksraj123) [reply activities/Abecedarium.activity/play.js]: Saw something similar in another activity so I went with it as I was not sure how to add localization to enyo activities. Figured it out, It has been fixed now.

**Comment by llaske** [comment]
Nice. Thanks.

**Pull request merged**